### PR TITLE
Add s2i to the func-utils image

### PR DESCRIPTION
**Pull request by matejvasek**
# Changes

- Incorporate `s2i`into the func-utils image.

### CODE REVIEW

File: Dockerfile.utils [hunk 1]
```diff
@@ -21,7 +21,8 @@ RUN apk add --no-cache socat tar
 
 COPY --from=builder /workspace/func-util /usr/local/bin/
 RUN ln -s /usr/local/bin/func-util /usr/local/bin/deploy && \
-    ln -s /usr/local/bin/func-util /usr/local/bin/scaffold
+    ln -s /usr/local/bin/func-util /usr/local/bin/scaffold && \
+    ln -s /usr/local/bin/func-util /usr/local/bin/s2i
```
Comment (matzew) [inline Dockerfile.utils:25]: how to we ensure this is up-to-date?

Is there any "harm" vendoring this extra binary?

File: pkg/pipelines/tekton/tasks.go [hunk 2]
```diff
@@ -303,7 +303,7 @@ spec:
       description: Digest of the image just built.
   steps:
     - name: generate
-      image: quay.io/boson/s2i:latest
```
Comment (matzew) [inline pkg/pipelines/tekton/tasks.go:306]: ugh, how old was that? :sweat_smile:

File: Dockerfile.utils (hunk 1 above)
Comment (matejvasek) [reply Dockerfile.utils:25]: Actually this ensures the s2i in the binary and the util image is the same which is nice. If there is any updated required (e.g. ther is CVE) we do it just in go.mod and it updates it both. What are disadvantages?

File: pkg/pipelines/tekton/tasks.go (hunk 2 above)
Comment (matejvasek) [reply pkg/pipelines/tekton/tasks.go:306]: Quite. Still much up to date that we use now in midstream.

File: Dockerfile.utils (hunk 1 above)
Comment (matzew) [reply Dockerfile.utils:25]: I was worried for carrying on CVEs, but looks like it is addressed

Comment (matejvasek) [reply Dockerfile.utils:25]: Well we basically already ship s2i binary. We use it as an library in `func` CLI.

Comment (matejvasek) [reply Dockerfile.utils:25]: >Well we basically already ship s2i binary. We use it as an library in func CLI.

So we are already on hook for keeping it up to date.

File: pkg/pipelines/tekton/tasks.go (hunk 2 above)
Comment (matejvasek) [reply pkg/pipelines/tekton/tasks.go:306]: `quay.io/boson/s2i:latest` is our image we build manually once in blue moon.

**Review by lkingland — APPROVED** [review]

**Pull request merged**